crates/wasmtime/src/runtime/profiling.rs

```diff
@@ -96,21 +106,42 @@ impl GuestProfiler {
     /// host code or functions from other modules will be omitted. See the
     /// "Security" section of the [`GuestProfiler`] documentation for guidance
     /// on what modules should not be included in this list.
-    pub fn new(module_name: &str, interval: Duration, modules: Vec<(String, Module)>) -> Self {
+    pub fn new(
+        module_name: &str,
+        interval: Duration,
+        modules: impl IntoIterator<Item = (String, Module)>,
+    ) -> Self {
         let zero = ReferenceTimestamp::from_millis_since_unix_epoch(0.0);
         let mut profile = Profile::new(module_name, zero, interval.into());
 
+        // past this point, we just need to think about modules as we pull out
+        // the disparate module information from components from this point forward.
```

Comment:
Nitpick for here and elsewhere: we generally try to capitalize sentences and such in comments.

Reply:
Super nits: Is "past this point, ... from this point forward" redundant? Maybe I should understand the context better. Also, while here, I'll point out the title has "suport" in it, probably a typo.

Reply:
Yeah, I noticed that as well when I was making updates.  Should be addressed with the fixup.